pkg/scheduler/volumebinder/volume_binder.go

```diff
@@ -43,27 +39,18 @@ func NewVolumeBinder(
 	storageClassInformer storageinformers.StorageClassInformer) *VolumeBinder {
 
 	return &VolumeBinder{
-		Binder:    persistentvolume.NewVolumeBinder(client, pvcInformer, pvInformer, storageClassInformer),
-		BindQueue: workqueue.NewNamed("podsToBind"),
+		// TODO: what is a good bind timeout value?
+		Binder: persistentvolume.NewVolumeBinder(client, pvcInformer, pvInformer, storageClassInformer, 10*time.Minute),
```

Comment:
Time consumption of provision for different storages can be vary, we'd better expose the timeout param to users

Reply:
How about we loop forever?  We early abort on many conditions, including:
- PVC or PV is deleted
- PVC selectedNode annotation got reset
- PV ClaimRef reset

We could also add one more check for if the Pod got deleted.

That way if the Pod gets stuck binding for some reason, there are various ways for the user to abort the process.

Reply:
The other concern is that this goroutine will be open as long as volume binding is pending.  Maybe we can have exponential backoff so all the goroutines won't wake up at the same time.